### PR TITLE
Update dependency slackclient to slack_sdk

### PR DESCRIPTION
**Pull request by ziegenberg**
##### SUMMARY

This PR updates the implementation of the slack backend for notifications to use the slack_sdk instead of the deprecated slackclient. This removes one `UPGRADE BLOCKER` from `requirements/requirements.in`. Unit tests are included.

According to the [official documentation](https://slack.dev/python-slackclient/):
> The [slackclient](https://pypi.org/project/slackclient/) PyPI project is in maintenance mode now and [slack-sdk](https://pypi.org/project/slack-sdk/) project is the successor.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - awx/main/notifications/slack_backend.py

##### AWX VERSION

```
19.4.1.dev93+g87a3638c32
```

##### ADDITIONAL INFORMATION

This needs manual testing before merging.

```

```

### CODE REVIEW

**Comment by ziegenberg** [comment]
Unit tests will be added.

**Comment by ziegenberg** [comment]
Unit tests have been added. Waiting for CI run.

**Comment by ziegenberg** [comment]
I can squash the commits for merging if necessary. I would do so after review.

File: awx/main/tests/unit/notifications/test_slack.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from unittest import mock
+from django.core.mail.message import EmailMessage
+
+import awx.main.notifications.slack_backend as slack_backend
+
+
+def test_send_messages():
+    with mock.patch('awx.main.notifications.slack_backend.WebClient') as slack_sdk_mock:
+        WebClient_mock = slack_sdk_mock.return_value
+        WebClient_mock.chat_postMessage.return_value = {'ok': True}
```
Comment (AlexSCorey) [inline awx/main/tests/unit/notifications/test_slack.py:12]: I think this is what I was missing in my own tests.

**Review by AlexSCorey — COMMENTED** [review]
This code looks good.  For the tests can we assert that it fails properly as well?

File: awx/main/notifications/slack_backend.py [hunk 2]
```diff
@@ -28,23 +29,23 @@ def __init__(self, token, hex_color="", fail_silently=False, **kwargs):
...
-                    logger.debug(ret)
-                    if ret['ok']:
+                        response = client.chat_postMessage(channel=r, as_user=True, text=m.subject)
+                    logger.debug(response)
+                    if response['ok']:
                         sent_messages += 1
                     else:
-                        raise RuntimeError("Slack Notification unable to send {}: {} ({})".format(r, m.subject, ret['error']))
-            except Exception as e:
+                        raise RuntimeError("Slack Notification unable to send {}: {} ({})".format(r, m.subject, response['error']))
```
Comment (ziegenberg) [inline awx/main/notifications/slack_backend.py:47]: @AlexSCorey do you want to test this code path as well?

**Comment by AlexSCorey** [comment]
> I can squash the commits for merging if necessary. I would do so after review.

Yes, please squash the commits.

File: awx/main/notifications/slack_backend.py (hunk 2 above)
Comment (ziegenberg) [reply awx/main/notifications/slack_backend.py:47]: I added a test for this as well.

**Comment by ziegenberg** [comment]
@AlexSCorey I squashed the commits and force-pushed. From my side, it's ready for review.

**Review by beeankha — APPROVED** [review]

**Comment by ziegenberg** [comment]
Is this just a flaky test or did I break the build?

**Comment by beeankha** [comment]
@ziegenberg you didn't break the build, this test failure is unrelated to the work you've done in this PR.  Currently getting a few more pairs of eyes on this before we merge your work.

**Comment by beeankha** [comment]
@ziegenberg thank you so much for this contribution! :tada:

**Comment by ziegenberg** [comment]
@beeankha you're welcome! there are three more from my side ready and waiting for review :)

**Pull request merged**